src/AngleStrategyVoterProxy.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.12;
+pragma experimental ABIEncoderV2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import {YearnAngleVoter} from "./YearnAngleVoter.sol";
+
+import "./interfaces/curve/ICurve.sol";
+import "./interfaces/Angle/IStableMaster.sol";
+import "./interfaces/Angle/IAngleGauge.sol";
+import "./interfaces/Uniswap/IUniV2.sol";
+
+library SafeVoter {
+    function safeExecute(
+        YearnAngleVoter voter,
+        address to,
+        uint256 value,
+        bytes memory data
+    ) internal {
+        (bool success, ) = voter.execute(to, value, data);
+        require(success);
+    }
+}
+
+contract AngleStrategyVoterProxy {
+    using SafeVoter for YearnAngleVoter;
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+    YearnAngleVoter public yearnAngleVoter;
+    address public constant angleToken = address(0x31429d1856aD1377A8A0079410B297e1a9e214c2);
+
+    // gauge => strategies
+    mapping(address => address) public strategies;
+    mapping(address => bool) public voters;
+    address public governance;
+
+    constructor(address _voter) public {
+        governance = address(0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52);
+        yearnAngleVoter = YearnAngleVoter(_voter);
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function approveStrategy(address _gauge, address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_gauge] = _strategy;
+    }
+
+    function revokeStrategy(address _gauge) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_gauge] = address(0);
+    }
+
+    function approveVoter(address _voter) external {
+        require(msg.sender == governance, "!governance");
+        voters[_voter] = true;
+    }
+
+    function revokeVoter(address _voter) external {
+        require(msg.sender == governance, "!governance");
+        voters[_voter] = false;
+    }
+
+    function lock(uint256 amount, uint256 unlockTime) external {
+        if (amount > 0) {
+            IERC20(angleToken).transfer(address(yearnAngleVoter), amount);
```

Comment:
I'd like to define the expected behavior of this function a bit better. Ideally we have an open `.lock()` function which allows anyone to max lock any ANGLE in the voter on our behalf.

Seems this design is expecting the proxy to carry an ANGLE token balance.  Usually all tokens kept will be in the voter, not proxy. Which means this logic will probably get skipped.

Reply:
I had the flow to leave the tokens in voter proxy rather than voter as I wanted to the voter to be a mere interface for veAngle. But as I think it's standard design, I changed it to Angle rewards be kept in the voter instead